utxonursery.go

```diff
@@ -964,101 +940,223 @@ func deserializeKidList(r io.Reader) ([]*kidOutput, error) {
 			if err == io.EOF {
 				break
 			} else {
 				return nil, err
 			}
 		}
-		kidOutputs = append(kidOutputs, kidOutput)
+		kidOutputs = append(kidOutputs, kid)
 	}
 
 	return kidOutputs, nil
 }
 
-// serializeKidOutput converts a KidOutput struct into a form
-// suitable for on-disk database storage. Note that the signDescriptor
-// struct field is included so that the output's witness can be generated
-// by createSweepTx() when the output becomes spendable.
-func serializeKidOutput(w io.Writer, kid *kidOutput) error {
+// CsvSpendableOutput is a SpendableOutput that contains all of the information
+// necessary to construct, sign, and sweep an output locked with a CSV delay.
+type CsvSpendableOutput interface {
+	SpendableOutput
+
+	// ConfHeight returns the height at which this output was confirmed.
+	// A zero value indicates that the output has not been confirmed.
+	ConfHeight() uint32
+
+	// SetConfHeight marks the height at which the output is confirmed in
+	// the chain.
+	SetConfHeight(height uint32)
+
+	// BlocksToMaturity returns the relative timelock, as a number of
+	// blocks, that must be built on top of the confirmation height before
+	// the output can be spent.
+	BlocksToMaturity() uint32
+
+	// OriginChanPoint returns the outpoint of the channel from which this
+	// output is derived.
+	OriginChanPoint() *wire.OutPoint
+}
+
+// babyOutput is an HTLC output that is in the earliest stage of upbringing.
+// Each babyOutput carries a presigned timeout transction, which should be
+// broadcast at the appropriate CLTV expiry, and its future kidOutput self. If
+// all goes well, and the timeout transaction is successfully confirmed, the
+// the now-mature kidOutput will be unwrapped and continue its journey through
+// the nursery.
+type babyOutput struct {
+	kidOutput
+
+	expiry    uint32
+	timeoutTx *wire.MsgTx
+}
+
+// makeBabyOutput constructs baby output the wraps a future kidOutput. The
+// provided sign descriptors and witness types will be used once the output
+// reaches the delay and claim stage.
+func makeBabyOutput(outpoint, originChanPoint *wire.OutPoint,
+	blocksToMaturity uint32, witnessType lnwallet.WitnessType,
+	htlcResolution *lnwallet.OutgoingHtlcResolution) babyOutput {
+
+	kid := makeKidOutput(outpoint, originChanPoint,
+		blocksToMaturity, witnessType,
+		&htlcResolution.SweepSignDesc)
+
+	return babyOutput{
+		kidOutput: kid,
+		expiry:    htlcResolution.Expiry,
+		timeoutTx: htlcResolution.SignedTimeoutTx,
+	}
+}
+
+// Encode writes the baby output to the given io.Writer.
+func (bo *babyOutput) Encode(w io.Writer) error {
+	var scratch [4]byte
+	byteOrder.PutUint32(scratch[:], bo.expiry)
+	if _, err := w.Write(scratch[:]); err != nil {
+		return err
+	}
+
+	if err := bo.timeoutTx.Serialize(w); err != nil {
+		return err
+	}
+
+	return bo.kidOutput.Encode(w)
+}
+
+// Decode reconstructs a baby output using the provide io.Reader.
+func (bo *babyOutput) Decode(r io.Reader) error {
```

Comment:
nit: Slight preference for the method name `Deserialize`.

Reply:
See above comment